src/Bundle/DependencyInjection/Configuration.php

```diff
@@ -115,8 +115,13 @@ private function addGridsSection(ArrayNodeDefinition $node): void
                                         ->arrayNode('options')
                                             ->performNoDeepMerging()
                                             ->variablePrototype()->end()
+                                            ->defaultValue([])
                                         ->end()
                                     ->end()
+                                    ->validate()
+                                        ->ifTrue(fn (array $config): bool => isset($config['options']['template']) && $config['type'] !== 'twig')
```

Comment:
This seems like an overreach that can plausibly cause conflicts with custom types.
Wouldn't it be better to reset the previous options once the `type` changes, we'd probably need to inject the logic somewhere inside the config merging mechanism though.